Add new content personalization docs page (#13890)
wagtail/wagtail: docs/advanced_topics/content_personalization.md

Make sure to use [`{% include_block %}`](streamfield_template_rendering) when rendering StreamField content so the block receives the parent context, including the `request` object used by `get_context`.

```python
class SegmentedContentBlock(StructBlock):
    content = RichTextBlock()
    segment = ChoiceBlock(
        choices=[
            ("all", "All visitors"),
            ("logged_in", "Logged-in users"),
            ("anonymous", "Anonymous visitors"),
        ],
        default="all",
    )

    def get_context(self, value, parent_context=None):
        context = super().get_context(value, parent_context)
        request = parent_context.get("request") if parent_context else None
        preview = getattr(request, "personalization_preview_segment", None)
        context["is_authenticated"] = (
            preview == "logged_in"
            if preview
            else (request.user.is_authenticated if request else False)

>         )

Reviewer: I think it's easier to understand with a more explicit syntax

```suggestion
     if preview:
        is_authenticated = preview == "logged_in"
     elif request:
        is_authenticated = request.user.is_authenticated
     else:
        is_authenticated = False
     context["is_authenticated"] = is_authenticated
```

Author: but it’s 2 lines longer! 😅

        return context

    class Meta:
        icon = "group"
        template = "blocks/segmented_content_block.html"
        preview_value = {
            "content": "<p>Welcome back! Exclusive content for members.</p>",
            "segment": "logged_in",
        }
        description = "Content targeted to specific audience segments"
        form_layout = BlockGroup(
            children=["content"],
            settings=["segment"],
        )
```

> This type of simple segmentation is simple to add to any StructBlock. See [](structblock_custom_order_and_grouping) for details on grouping and ordering StructBlock fields, for scenarios where blocks are already more complex.

Reviewer: Suggestion:

```suggestion
This type of simple segmentation can be added to any StructBlock. See [](structblock_custom_order_and_grouping) for details on grouping and ordering StructBlock fields, for scenarios where the block has a more complex structure.
```
